extensions/blocks/podcast-player/podcast-player.php

```diff
@@ -179,3 +208,69 @@ function setup_tracks_callback( \SimplePie_Item $episode ) {
 
 	return $track;
 }
+
+/**
+ * Build an array with CSS classes and inline styles defining the colors
+ * which will be applied to player either in the editor-canvas
+ * as well as in the front-end side.
+ *
+ * @param  array $attributes block attributes.
+ * @return array Colors CSS classes and inline styles.
+ */
+function get_colors( $attributes ) {
+	$colors = array(
+		'css_classes'      => array(),
+		'css_text_classes' => array(),
+		'style'            => '',
+		'class_name'       => '',
+		'text_class_name'  => '',
+	);
+
+	// Text color.
+	$has_named_text_color  = isset( $attributes['textColor'] );
+	$has_custom_text_color = isset( $attributes['customTextColor'] );
+
+	// If has text color.
+	if ( $has_custom_text_color || $has_named_text_color ) {
+		// Add has-text-color class.
+		$colors['css_classes'][]      = 'has-text-color';
+		$colors['css_text_classes'][] = 'has-text-color';
+	}
+
+	if ( $has_named_text_color ) {
+		// Add the color class.
+		$colors['css_classes'][]      = sprintf( 'has-%s-color', $attributes['textColor'] );
+		$colors['css_text_classes'][] = sprintf( 'has-%s-color', $attributes['textColor'] );
```

Comment:
Are both of these necessary? I was trying to read through all of this and it seems a bit complicated to follow the whole flow.

Reply:
yes, we are applying these CSS classes to different HTML elements. The css_classes contains all CSS classes while `css_text_classes` contains only the CSS class about text.